### PR TITLE
Update diamond conda version to 0.7.10

### PR DESCRIPTION
**Pull request by yesimon**
To address #279

### CODE REVIEW

File: tools/diamond.py [hunk 1]
```diff
@@ -27,7 +28,9 @@ class Diamond(tools.Tool):
 
     def __init__(self, install_methods=None):
         if not install_methods:
-            install_methods = [DownloadAndBuildDiamond(URL, os.path.join(DIAMOND_DIR, 'bin', 'diamond'))]
+            install_methods = [
+                DownloadAndBuildDiamond(URL, os.path.join(DIAMOND_DIR, 'bin', 'diamond')),
+                tools.CondaPackage("diamond", version=CONDA_VERSION)]
```
Comment (dpark01) [inline tools/diamond.py]: Is there a reason that Conda should only be tried if download-and-build fails? Most tools specify these installers in the opposite order for the sake of speed-optimization.

Comment (yesimon) [inline tools/diamond.py]: You are right. This is not ready yet because the `CondaPackageVersion` spec is in another PR.

**Comment by dpark01** [comment]
Just merged in the other PR. Rebase on master and flip the tool installer order?

**Comment by yesimon** [comment]
Lots of conda commands return nonzero exit codes which were already handled previously so we shouldn't use `check=True`.

**Comment by dpark01** [comment]
Ok, go ahead and merge this asap once it passes. Then tag v1.4.1.

**Comment by dpark01** [comment]
This is a bit disconcerting. Not sure why the shell script returns exit code 1 when invoked with subprocess.run on Py3.5 but appears to exit 0 with subprocess.check_output on 3.4 and 2.7. Not sure who's right / which one is the bug.

I'm inclined to harmonize it by removing the try: from subprocess import run statement and always use your custom run method even if subprocess.run exists. While that feels backwards, the real goal is not to re-implement something from 3.5 accurately, but to create a method that works in our own use cases consistently.

**Pull request merged**